vignettes/gsmExtensions.Rmd

```diff
@@ -0,0 +1,422 @@
+---
+title: "gsm Extensions"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{gsmExtensions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Overview
+
+This vignette describes how to extend `gsm` by creating new "modules", including metrics, reports and shiny apps that can be run using the standard `gsm` pipeline described in these vignettes (`vignette("DataAnalysis")`, `vignette("DataReporting")`). As shown in the `vignette("DataAnalysis")`, the `gsm` data pipeline can be used to capture a monitoring 'snapshot' for a study that includes a variety of "modules" including [metrics](https://github.com/Gilead-BioStats/gsm/tree/dev/inst/workflow/metrics/kri0001.yaml), [reports](https://github.com/Gilead-BioStats/gsm/blob/dev/inst/workflow/reports/report_kri_site.yaml) and [apps](TODO: Add this link).Some core modules are included in the `gsm` package, while others can be added as extensions. 
+
+This vignette provide detailed specifications for creating new modules, a description of the directory structure for the `yaml` workflows that comprise a module pipeline, and links to resources that can be used to configure study-level gsm pipelines that utilize these `gsm` extensions.
+
+# Module Configuration
+
+`gsm` modules will typically be part of an R package and have a YAML configuration file - `{module_name}.yaml` - saved under `\inst\workflow\4_modules`. Module config files include 3 key properties: 
+
+  -  `meta`: (required) Report metadata used for centralized library and fields referenced in `steps` section of workflow.
+    - This is not required in the `{report_name}.yaml` if the report is entirely comprised of `metric`s that have their own `yaml` configuration files. 
```

Comment:
I don't fully understand this line. Delete?

Reply:
yeah- this no longer applies. deleting.